include/db/sstable_builder.hh

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "common.hh"
+#include "db/sstable_files.hh"
+
+namespace dibibase::db {
+
+class DIBIBASE_PUBLIC SSTableBuilder {
```

Comment:
@hassanelshazly I don't think she meant the Builder Design Pattern. I think even a Factory is overkill here. It's not like we are choosing among polymorphic types. This is the SSTable itself, it just depends on an I/O layer beneath it to get the files. This will change to a better structure in the future I'm sure.